client/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -1,38 +1,147 @@
+  // const { initSocket } = useSocket();
+
+  // const history = useHistory();
+
   const classes = useStyles();
+  const dashOptions = ['EVENT TYPES', 'SCHEDULED EVENTS'];
+  const [dashOptionSelected, setDashOption] = React.useState(dashOptions[0]);
+  const schedOptions = ['UPCOMING', 'PENDING', 'PAST'];
+  const [schedSelect, setSchedSelect] = React.useState(schedOptions[0]);
+  const meetingOptions = [15, 30, 45];
+
+  // useEffect(() => {
+  //   initSocket();
+  // }, [initSocket]);
 
-  const { loggedInUser } = useAuth();
-  const { initSocket } = useSocket();
+  // if (loggedInUser === undefined) return <CircularProgress />;
+  // if (!loggedInUser) {
+  //   history.push('/login');
+  //   // loading for a split seconds until history.push works
+  //   return <CircularProgress />;
+  // }
 
-  const history = useHistory();
+  const createOptions = (
+    choices: string[],
+    stateVar: string,
+    stateUpdate: React.Dispatch<React.SetStateAction<string>>,
+  ) => {
+    const output = [];
+    for (let i = 0; i < choices.length; i++) {
+      output.push(
+        <Button
+          key={choices[i]}
+          className={stateVar === choices[i] ? `${classes.option} ${classes.selected}` : classes.option}
+          onClick={function () {
+            stateUpdate(choices[i]);
+          }}
+        >
+          {choices[i]}
+        </Button>,
+      );
+    }
+    return output;
+  };
 
-  useEffect(() => {
-    initSocket();
-  }, [initSocket]);
+  const displayMeetOptions = (options: number[]) => {
+    const colors = ['purple', 'green', 'orange'];
+    const output = [];
+    for (let i = 0; i < options.length; i++) {
+      output.push(
+        <ScheduleOption key={`meeting option ${i}`} schedTime={options[i]} colour={colors[i % colors.length]} />,
+      );
+    }
+    return output;
+  };
 
-  if (loggedInUser === undefined) return <CircularProgress />;
-  if (!loggedInUser) {
-    history.push('/login');
-    // loading for a split seconds until history.push works
-    return <CircularProgress />;
-  }
+  const populateSchedEvent = () => {
+    const output = [];
+    const events = [];
+    if (schedSelect === schedOptions[0]) {
+      // Need to fill in the code where all of the appropriate events are getting populated.
+      // Note that this is just a dummy data and this code needs to change depending on the structure
+      // of the data recieved from the database.
+      events.push({ name: 'Alan', duration: 60 });
+      events.push({ name: 'Rickman', duration: 30 });
+    } else if (schedSelect === schedOptions[1]) {
+      // Need to fill in the code where all of the appropriate events are getting populated.
+    } else if (schedSelect === schedOptions[2]) {
+      // Need to fill in the code where all of the appropriate events are getting populated.
+    }
+    if (events.length > 0) {
+      // Create list or div elements for each one of the events and populate the output.
```

Comment:
This code is pretty self-explanatory which means no need to comment the behaviour, good job!